stdpopsim/slim_engine.py

```diff
@@ -214,34 +286,31 @@
 
 
 def slim_makescript(
-        script_file,
-        trees_file,
-        demographic_model,
-        samples,
-        Q,
-        check_coalescence,
-        mutation_rate,
-        generation_time,
-        population_configurations,
-        recombination_map,
-        migration_matrix,
-        demographic_events,
-        verbosity):
+        script_file, trees_file,
+        demographic_model, contig, samples,
+        Q, check_coalescence, verbosity):
 
+    generation_time = demographic_model.generation_time
     if generation_time <= 0:
         raise Exception(f"generation_time={generation_time} is invalid")
 
+    recombination_map = contig.recombination_map
     if len(recombination_map.get_positions()) > 2:
         raise Exception("recombination_map not supported")
 
-    pop_names = [pc.metadata["id"] for pc in population_configurations]
+    pop_names = [pc.metadata["id"] for pc in demographic_model.population_configurations]
+
+    # Reassign event times according to integral SLiM generations.
+    # This collapses the time deltas used in HomSap/AmericanAdmixture_4B11.
+    for event in demographic_model.demographic_events:
+        event.time = int(event.time / Q) * Q
```

Comment:
Why not ``event.time = round(event.time)``? That seems fairly well specified and easy to document

Reply:
Although I'd prefer if we errored out with non-integral times --- seems like we're storing up gotchas for later here, and the original source of the problem could be resolved.

Reply:
Simply rounding event.time doesn't quite cut it. This ``Q`` parameter rescales several things, one of which is the generation times (So 1 generation in SLiM will correspond to Q actual generations). I've rounded the value that actually needs to be an integer (``event.time/Q``)